### PR TITLE
tintColor for Android ImageView

### PR DESCRIPTION
**Pull request by m1ga**
https://jira.appcelerator.org/browse/TIMOB-19912

Adding tintColor to Android ImageView

```
$.img.setTintColor("#FF0000");
$.img.tintColor = "#F0FF0F";
```

### CODE REVIEW

**Comment by hansemannn** [comment]
Hi @m1ga, are you still working on this PR? Just noticed that the CI fails. Thanks!

**Comment by m1ga** [comment]
Hi,
not sure why CI fails. This is the error:
~~~
[INFO]  file:/data/data/com.appcelerator.testApp.testing/cache/
[INFO]  Started: multiLingualFilename
[ERROR] TiFileProxy: (KrollRuntimeThread) [1854,13780] IOException encountered
[ERROR] TiFileProxy: java.io.IOException: read only
[ERROR] TiFileProxy: 	at org.appcelerator.titanium.io.TiResourceFile.write(TiResourceFile.java:96)
[ERROR] TiFileProxy: 	at org.appcelerator.titanium.TiFileProxy.write(TiFileProxy.java:286)
[ERROR] TiFileProxy: 	at org.appcelerator.kroll.runtime.v8.V8Function.nativeInvoke(Native Method)
[ERROR] TiFileProxy: 	at org.appcelerator.kroll.runtime.v8.V8Function.callSync(V8Function.java:57)
[ERROR] TiFileProxy: 	at org.appcelerator.kroll.runtime.v8.V8Function.call(V8Function.java:43)
[ERROR] TiFileProxy: 	at ti.modules.titanium.TitaniumModule$Timer.run(TitaniumModule.java:152)
[ERROR] TiFileProxy: 	at android.os.Handler.handleCallback(Handler.java:739)
[ERROR] TiFileProxy: 	at android.os.Handler.dispatchMessage(Handler.java:95)
[ERROR] TiFileProxy: 	at android.os.Looper.loop(Looper.java:135)
[ERROR] TiFileProxy: 	at org.appcelerator.kroll.KrollRuntime$KrollRuntimeThread.run(KrollRuntime.java:118)
~~~

and it stops with
~~~
[INFO]  {"date":"2016-01-24T16:33:09.678Z","results":[{"state":"passed","duration":136,"suite":"Titanium.UI.Label","title":"text"},{"state":"passed","duration":25,"suite":"Titanium.UI.Label","title":"textAlign"},{"state":"passed","duration":36,"suite":"Titanium.UI.Label","title":"verticalAlign"},{"state":"passed","duration":55,"suite":"Titanium.UI.ImageView","title":"image"},{"state":"passed","duration":158,"suite":"filesystem","title":"optionalArgAPIs"},{"state":"passed","duration":1116,"suite":"filesystem","title":"readWriteText"},{"state":"passed","duration":55,"suite":"filesystem","title":"blobNativeFile"},{"state":"passed","duration":28,"suite":"filesystem","title":"blobFile"},{"state":"passed","duration":30,"suite":"filesystem","title":"dotSlash"},{"state":"passed","duration":326,"suite":"filesystem","title":"appendStringTest"},{"state":"passed","duration":2567,"suite":"filesystem","title":"appendBlobTest"},{"state":"passed","duration":2141,"suite":"filesystem","title":"appendFileTest"},{"state":"passed","duration":194,"suite":"filesystem","title":"fileStreamBasicTest"},{"state":"passed","duration":60,"suite":"filesystem","title":"fileStreamWriteTest"},{"state":"passed","duration":71,"suite":"filesystem","title":"fileStreamAppendTest"},{"state":"passed","duration":259,"suite":"filesystem","title":"fileStreamPumpTest"},{"state":"passed","duration":72,"suite":"filesystem","title":"fileStreamWriteStreamTest"},{"state":"passed","duration":110,"suite":"filesystem","title":"fileStreamResourceFileTest"},{"state":"passed","duration":56,"suite":"filesystem","title":"fileStreamTruncateTest"},{"state":"passed","duration":47,"suite":"filesystem","title":"fileMove"},{"state":"passed","duration":205,"suite":"filesystem","title":"tempDirTest"},{"state":"passed","duration":759,"suite":"filesystem","title":"emptyFile"},{"state":"passed","duration":889,"suite":"filesystem","title":"fileSize"},{"state":"passed","duration":8,"suite":"filesystem","title":"mimeType"},{"state":"
[INFO]  !TEST_RESULTS_STOP!
undefined:1
te":"passed","duration":8,"suite":"filesystem","title":"mimeType"},{"state":" 

                                                                              
SyntaxError: Unexpected end of input
    at Object.parse (native)
    at parseAndroidTestResults (/Users/travis/build/appcelerator/titanium_mobile/support/ti_mocha_script/test.js:260:29)
    at async.series.callback.iosResults (/Users/travis/build/appcelerator/titanium_mobile/support/ti_mocha_script/test.js:324:5)
    at /Users/travis/build/appcelerator/titanium_mobile/node_modules/async/lib/async.js:713:13
    at iterate (/Users/travis/build/appcelerator/titanium_mobile/node_modules/async/lib/async.js:262:13)
    at /Users/travis/build/appcelerator/titanium_mobile/node_modules/async/lib/async.js:274:29
    at /Users/travis/build/appcelerator/titanium_mobile/node_modules/async/lib/async.js:44:16
    at /Users/travis/build/appcelerator/titanium_mobile/node_modules/async/lib/async.js:718:17
    at /Users/travis/build/appcelerator/titanium_mobile/node_modules/async/lib/async.js:167:37
    at ChildProcess.<anonymous> (/Users/travis/build/appcelerator/titanium_mobile/support/ti_mocha_script/test.js:217:4)
npm ERR! Test failed.  See above for more details.
~~~

I wasn't sure if it has something todo with my patch or something on CI.

The Jira ticket is at: https://jira.appcelerator.org/browse/TIMOB-19912

**Comment by hansemannn** [comment]
Re-triggered travis..Lets see

**Comment by m1ga** [comment]
all green :)

**Comment by hansemannn** [comment]
Great! So is the PR finished and can be reviewed by the Android team?

**Comment by m1ga** [comment]
yes, it can be reviewed. Thanks

File: android/modules/ui/src/java/ti/modules/titanium/ui/WebViewProxy.java [hunk 1]
```diff
@@ -214,6 +218,23 @@ public void setBasicAuthentication(String username, String password)
 	}
 
 	@Kroll.method @Kroll.setProperty
+	public void setHeader(HashMap d)
```
Comment (hansemannn) [inline android/modules/ui/src/java/ti/modules/titanium/ui/WebViewProxy.java]: How are these changes related to the tintColor?

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java [hunk 2]
```diff
@@ -388,6 +390,13 @@ private boolean mightBeHtml(String url)
 		}
 	}
 
+	public void setHeader(HashMap items){
+			Map<String, String> map = items;
+			for(Map.Entry<String, String> item : map.entrySet()){
```
Comment (hansemannn) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java]: Spacing

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java [hunk 3]
```diff
@@ -459,7 +468,11 @@ public void setUrl(String url)
 			getWebView().getSettings().setLoadWithOverviewMode(true);
 		}
 		isLocalHTML = false;
-		getWebView().loadUrl(finalUrl);
+		if (extraHeaders.size()>0){
```
Comment (hansemannn) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiUIWebView.java]: Spacing

File: apidoc/Titanium/UI/ImageView.yml [hunk 4]
```diff
@@ -230,22 +230,29 @@ properties:
...
     summary: Run the animation in reverse.
     type: Boolean
     default: false
 
+  - name: tintColor
+    summary: The view's tintColor.
+    type: String
+    default: null
+    platforms: [android, iphone, ipad, mobileweb]
+    since: { android: "5.2.0", iphone: "3.1.3", ipad: "3.1.3", mobileweb: "4.1.0"}
```
Comment (hansemannn) [inline apidoc/Titanium/UI/ImageView.yml]: Change android to 5.4.0

File: apidoc/Titanium/UI/WebView.yml [hunk 5]
```diff
@@ -141,17 +141,33 @@ excludes:
 since: "0.8"
 methods:
 
+- name: setHeader
```
Comment (hansemannn) [inline apidoc/Titanium/UI/WebView.yml]: Doesn't like tintColor related (see above)

**Comment by hansemannn** [comment]
@m1ga Left some comments. If possible, please add implement your changes with Eclipse/Xcode again, to ensure the spacing is correctly. Your PR includes lots of blank lines that are replaced (see the red and green lines). That makes it hard to diff a PR later and produces possible missleading merge conflicts for newer PR's. Thank you!

**Comment by hansemannn** [comment]
In Addition, can you think of a way to support the tintColor on the `Ti.UI.Button.image` property as well?

**Comment by m1ga** [comment]
@hansemannn I found a way to ignore the spaces (atom's fault ;) ) now. Should I create a new PR?

**Comment by hansemannn** [comment]
I would love it! But if it blocks your productivity, I'd be fine with this one. Thank you!

**Comment by m1ga** [comment]
new clean PR: https://github.com/appcelerator/titanium_mobile/pull/7877
`Ti.UI.Button.image` will be a different PR
